services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy

```diff
@@ -117,7 +125,46 @@ class IntentionalRerouteSpec extends BaseSpecification {
         and: "Remove the flow, restore bandwidths on ISLs, reset costs"
         Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
         flowHelper.deleteFlow(flow.id)
-        [thinIsl, islUtils.reverseIsl(thinIsl)].each { database.revertIslBandwidth(it) }
+        [thinIsl, thinIsl.reversed].each { database.revertIslBandwidth(it) }
+    }
+
+    def "Intentional flow reroute is not causing any packet loss"() {
+        given: "An unmetered flow going through unpreferable path(reroute potential)"
+        //will be available on virtual as soon as we get the latest iperf installed in lab-service images
+        requireProfiles("hardware")
+        def flow = flowHelper.randomFlow(topology.activeTraffGens[0].switchConnected,
+                topology.activeTraffGens[1].switchConnected)
+        flow.maximumBandwidth = 0
+        flow.ignoreBandwidth = true
+        flowHelper.addFlow(flow)
+        def path = northbound.getFlowPath(flow.id)
+        pathHelper.makePathUnpreferable(path)
+
+        when: "Start traffic examination"
+        def traffExam = traffExamProvider.get()
+        def bw = 100000 // 100 Mbps
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildBidirectionalExam(flow, bw)
+        [exam.forward, exam.reverse].each { direction ->
+            def resources = traffExam.startExam(direction, true)
+            direction.setResources(resources)
+        }
+
+        and: "While traffic flow is active, request a flow reroute"
+        [exam.forward, exam.reverse].each { assert !traffExam.isFinished(it) }
+        northbound.rerouteFlow(flow.id)
+
+        then: "Flow is rerouted"
+        northbound.getFlowPath(flow.id) != path
```

Comment:
Is a waiter needed here?

Reply:
as far as I understand this operation is synchronized (actual rules installation is async, but db is updated before reroute response is returned). I didn't experience any problems here

Reply:
moreover, the updated path is already returned as a response on L184

Reply:
Ok.